### <b><h1><center><font color = 'silver'>Aplikacja mobilna - Twoja szansa na sukces !</font></center></h1></b1>
### <h1><center>✨📱📱📱✨</center></h1> 

Źródło : https://www.kaggle.com/datasets/lava18/google-play-store-apps?select=googleplaystore.csv

<b>Założenia:</b>
1. W celu weryfikacji 'popularności' aplikacji przeprowadzono badanie gęstości występowania aplikacji względem oceny / liczby recenzji i liczby pobrań dla poszczególbych cech. Zauważalna różnica w rozkładzie jest czynnikiem wartym rozważenia pod kątem 'popularności'.
2. Wyznacznikiem 'popularności' pod kątem cechy 'aplikacja płatna/darmowa' uznano stosunek ilości pobrań aplikacji darmowej do płatnej.
3. W pkt 2 skupiono się na ilości pobrań, gdyż jest to pierwsza główna skłądowa do analizy popularności aplikacji mobilnych. Jeśli aplikacja jest pobierana, uznaje się że jest znana, a zatem bardziej popularna.
4. Przeprowadzono standaryzację i kategoryzację cechy 'rozmiar aplikacji'. Na tej podstawie pogrupowano aplikacje w klasy rozmiarowe.
5. <font color='green'><b> Finalnym zestawieniem popularności aplikacji jest scoreboard utworzony na bazie rankingu częstości występowań danych cech (Kategoria | Odbiorca | Rozmiar | Aplikacja płatna/ darmowa). Scoreboard został przeprowadzony na kwantylu 0.8 w wystąpieniach, aby odciąć granicę popularnych aplikacji od ogółu. </b></font> 
6. <font color='gold'><b> Na bazie powyższej analizy przygotowano aplikację, która umożliwi porównać czy nowa aplikacja mobilna ma szansę być popularna i finalnie odnieść sukces. </b></font>



<b> Pod podanym źródłem znajdują się dwa pliki:
1. 'googleplaystore' - baza aplikacji zawierająca informacje: nazwa | typ | rozmiar | kategoria | content rating | ocena | il. recenzji | ilość pobrań
2. 'googleplaystor_user_reviews' - baza recenzji, przydatna do analizy tekstu. Baza ta nie została wykorzystana dla poniższej analizy, natomiast stanowi punkt dalszej analizy i być może będzie podstawą do przewidywania recenzji nowej aplikacji </b>

In [2]:
from project_functions import *
pass

clean_data()

App                Type  Size       Category      Content Rating  Rating  Reviews  Installs  Size mb  Size category
0  +Download 4 Instagram Twitter  Free   22M              SOCIAL     Everyone      4.5    40467    1000000  22.000        2.0     
1     - Free Comics - Comic Apps  Free  9.1M              COMICS   Mature 17+      3.5      115      10000   9.100        0.0     
2                             .R  Free  203k               TOOLS     Everyone      4.5      259      10000   0.198        0.0     
3                         /u/app  Free   53M       COMMUNICATION   Mature 17+      4.7      573      10000  53.000        5.0     
4                         058.ba  Free   14M  NEWS_AND_MAGAZINES     Everyone      4.4       27        100  14.000        1.0

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7025 entries, 0 to 8195
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             7025 non-null   object 
 1   Type            7025 non-null   object 
 2   Size            7025 non-null   object 
 3   Category        7025 non-null   object 
 4   Content Rating  7025 non-null   object 
 5   Rating          7025 non-null   float64
 6   Reviews         7025 non-null   int64  
 7   Installs        7025 non-null   int64  
 8   Size mb         7025 non-null   float64
 9   Size category   7025 non-null   float64
dtypes: float64(3), int64(2), object(5)
memory usage: 603.7+ KB
None
No of duplicates: 0


<center><ins> W celu analizy usunieto duplikaty, puste wartości oraz dane dla rozmiaru, gdzie nie został on określony. Finalnie bazowano na <font color='pink'>7025 rekordach i 4 głównych cechach składowych </font> ('Type'/ 'Content Rating'/ 'Size mb' / 'Category') </ins></center>

In [ ]:
means()

In [ ]:
correlation()

<center><b> Na podstawie powyższej analizy korelacji zauważa się, że wszystkie analizowane cechy nie są współzależne. Największa korelacja występuje pomiędzy ilością recenzji, a liczbą pobrań co jest dość oczywiste - ilość recenzji po części wynika z ilości pobrań, natomiast nie oznacza to że każda pobrana aplikacja jest recenzowana </b></center>

<font color = 'orange'><b> Na podstawie analizy gęstości dla cech głównych składowych wymienionych wyżej, wnioskuje się iż kryterium doboru do rankingu w scoreboard będzie stosunek ilosci pobrań aplikacji darmowej do aplikacji płatnych. W przypadku pozostałych cech uznano, że do scoreboard wykorzysta się q = 0.8</b></font>

In [ ]:
type_charts_installs()

Powyższy wykres pokazuje, że rozkąłdy są zuepłeni inne. Nie wnioskujemy na podstawie powyższego wykresu jakie są różnice - badamy zależność dalej.

In [ ]:
mean_ratio()

In [ ]:
chart_free_type_no()

<b>Badanie cechy 'Category' na wykresie gęśtości (badano w celu porównania zależności i znalezienia specyficznego przypadku):</b>

In [ ]:
chart_density_app()

In [ ]:
chart_density_rating_app()

<b>Zestawienie ilości aplikacji w stosunku do liczby pobrań dla danej kategorii:</b>

In [ ]:
installs_category_no()

<b>Badane cechy 'Size mb' - statystyka obrazująca ilość aplikacji bez przypisanej wartosci :</b>

In [ ]:
chart_size_data_no_data ()

<b> Statystyka tylko dla wybranych do analizy aplikacji: </b>

In [ ]:
statistic_type()

# <b> Tworzenie scorecardu - opis metodyki: </b>

1. Wykonano zestawienie ilości występowań danej kategorii aplikacji ograniczonego o q = 0.8
2. Wykonano zestawienie ilości występowań danej grupy odbiorców ograniczonej o q = 0.8
3. Wykonano zestawieie ilości występowań danej klasy rozmiaru aplikacji (podział na klasy jako przedziały wartości rozmiarów) ograniczone o q =0.8
4. Dla każdego zestawienia oblicozno punktację zgodnie z definicją: max(ilość wystąpień) / (ilość wystąpień) = score
5. Punktacje dla kategorii, rozmiaru oraz odbiorcó dodano do siebie
6. Aby wyróżnić znaczenie apliakcji darmowej jako miarę popularności - powyższą sumę przemnożono przez stosunek ilości pobran aplikacji darmowej do płatnej.

<h1><center><b><font color='green'> SCORECARD </font></b></center></h1>

## <center>🥇🥈 🥉</center>

In [ ]:
scoreboard()

<h1><center><b><font color ='gold'>Na podstawie powyższego zestawienia najpopularniejszymi aplikacjami są te, które są z dziedziny gamingowej dedykowane dla każdego użytkownika, będące bezpłatne. Zazwyczaj te aplikacje mają ok 50 Mb rozmiaru.</font></b></center></h1>

<h1><center>Teraz Twoja kolej! Spróbuj wbić się w rynek, sprawdź czy aplikacja, którą chesz utworzyć ma szansę powodzenia ! 😉😉 🔥🔥🔥</center></h1>

In [3]:
app()

ToggleButtons(options=('Free', 'Paid'), value='Free')

Button(button_style='primary', description='CALCULATE SCORE', layout=Layout(height='40px', width='25%'), style…

Output()